snapcraft/storeapi/__init__.py

```diff
@@ -372,7 +374,8 @@ def get_package(self, snap_name, channel, arch=None):
     def get(self, url, headers=None, params=None, stream=False):
         if headers is None:
             headers = {}
-        headers.update({'Authorization': _macaroon_auth(self.conf)})
+        with contextlib.suppress(errors.InvalidCredentialsError):
```

Comment:
We cannot assume all requests require a login, can we? We already have the with login guard on every call that requires one. If you are not logged in there is no need to _macaroon_auth